---
title: "Figure_5a"
author: "David Bradley"
date: '2023-07-31'
output: html_document
---

```{r}

# Run Kinannote on all of the proteomes

metazoa_genomes <- list.files()[grep('.gz',list.files())]
metazoa_names <- rapply(strsplit(metazoa_genomes,split='_'),function(x)x[1])

for (i in 1:length(metazoa_genomes)) {
  
  print(i)
  
  tag <- metazoa_names[i]
  
  mkdir_command <- paste('mkdir',tag)
  system(mkdir_command)
  
  proteome <- metazoa_genomes[i]
  
  perl_com <- paste('perl Kinannote_1.0.pl -m -c',tag,proteome) # N.B. The -m tag means 'metazoa' and means that the predictions can include tyrosine kinases
  system(perl_com)
  
  tag_files <- list.files()[grep(tag,list.files())]
  tag_files <- setdiff(tag_files,proteome)
  
  mv_com <- paste('mv',tag_files,tag)
  
  for (j in 1:length(mv_com)) {
    system(mv_com[j])
  }
  
}

```

# Retrive the number of tyrosine kinases for each proteome

```{r}

metazoa_genomes <- list.files()[grep('.gz',list.files())]
metazoa_names <- rapply(strsplit(metazoa_genomes,split='_'),function(x)x[1])

TK_num_vec <- NULL

for (i in 1:length(metazoa_genomes)) {
  
  print(i)
  
  tag <- metazoa_names[i]
  proteome <- metazoa_genomes[i]
  
  tag_proteome <- paste(tag,proteome,sep='/')
  tag_proteome <- gsub('.gz','.names',tag_proteome)
  
  kin_list <- read.table(tag_proteome) 
  kin_class <- rapply(strsplit(kin_list[,2],split='/'), function(x)x[1])
  TK_num <- length(which(kin_class == 'TK'))
  TK_num_vec <- c(TK_num_vec, TK_num)
  
}

# Add zeros corresponding to the 6 fungal species (which have no tyrosine kinases)

names(TK_num_vec) <- metazoa_names
TK_num_vec <- c(0,0,0,0,0,0,TK_num_vec)

```

# Y content of fungal proteomes

```{r}

fungi_proteomes <- list.files('') # directory containing the fungal proteomes
fungi_proteomes <- fungi_proteomes[grep('.gz',fungi_proteomes)]
fungi_names <- rapply(strsplit(fungi_proteomes,split='_'), function(x) x[1])

Y_freq_vec <- NULL

for (i in 1:length(fungi_proteomes)) {
  
  fungi_proteome <- fungi_proteomes[i]
  
  fas <- seqinr::read.fasta(fungi_proteome,seqtype='AA')
  freq_table <- table(unname(unlist(fas)))/sum(table(unname(unlist(fas)))) # relative frequencies for each amino acid
  freq_Y <- unname(freq_table[names(freq_table) == 'Y'])
  
  Y_freq_vec <- c(Y_freq_vec,freq_Y)
  
}

names(TK_num_vec)[1:6] <- fungi_names
names(Y_freq_vec) <- fungi_names

```

# Y content of metazoa genomes

```{r}

metazoa_genomes <- list.files()[grep('.gz',list.files())]
metazoa_names <- rapply(strsplit(metazoa_genomes,split='_'),function(x)x[1])

for (i in 1:length(metazoa_genomes)) {
  
  print(i)
  
  proteome <- metazoa_genomes[i]
  
  fas <- seqinr::read.fasta(proteome,seqtype='AA')
  freq_table <- table(unname(unlist(fas)))/sum(table(unname(unlist(fas)))) # relative frequencies for each amino acid
  freq_Y <- unname(freq_table[names(freq_table) == 'Y'])
  
  Y_freq_vec <- c(Y_freq_vec,freq_Y)
  
}

names(Y_freq_vec)[7:length(Y_freq_vec)] <- metazoa_names

```

# No significant correlation

```{r}

plot(TK_num_vec,Y_freq_vec)
cor.test(TK_num_vec,Y_freq_vec)

## ggplot the results

reg_df <- data.frame(TK_num_vec[-9],Y_freq_vec[-9]) # remove rotifers that have conflicting results between species
colnames(reg_df) <- c('Kinase','Y')

library(ggrepel)
library(ggplot2)
library(ggtext)

row_nom <- rapply(strsplit(rownames(reg_df),split=''), function(x) paste(x[1],paste(x[2:length(x)],collapse='')))
substr(row_nom,3,3) <- tolower(substr(row_nom,3,3))
row_nom <- paste(paste(substr(row_nom,1,1),'.',sep=''),rapply(strsplit(row_nom,split=' '), function(x) x[2]))
row_nom[1] <- 'A. macrogynus' # manually modify species names
row_nom[25] <- 'T. adhaerans' # manually modify species names
rownames(reg_df) <- row_nom

fungi_df <- reg_df[1:6,]
metazoa_df <- reg_df[7:nrow(reg_df),]

p <- ggplot(reg_df, aes(x=Kinase, y=Y)) + geom_point()
p <- p+geom_label_repel(data=metazoa_df,aes(label = rownames(metazoa_df)),size=1.7, label.size=NA, fill=NA, color=colors()[131])
p <- p+geom_label_repel(data=fungi_df,aes(label = rownames(fungi_df)),size=1.7, label.size=NA, fill=NA, color=colors()[142])
p <- p+geom_smooth(method=lm, se=FALSE, col='black', lwd=0.6)
p <- p+theme_bw() + theme(text=element_text(face="plain", size=15), panel.border = element_rect(color="black", size=1.2, linetype="solid"))+theme(legend.position="none")
p <- p+theme(axis.text=element_text(size=9),axis.title.x=element_text(size=9,face="bold"),axis.title.y=element_text(size=9,face="bold"),plot.title=element_text(size=18,face='bold'))
p <- p+theme(legend.position="none")
p <- p + ylab("Proteome Y (relative frequency)") + xlab("No. of predicted Y kinases") + ggtitle('')
p <- p + geom_richtext(x = 128, y = 0.0384, label='r<sup>2</sup> = 0.006', check_overlap = TRUE, size=3.0, label.color = NA)
ggsave(file='Figure_5a.pdf', plot=p, width=4, height=3)

# write out the table

write.table(reg_df,'Figure_5a.txt',col.names=T,row.names=T,quote=F)
write.csv(reg_df,'Figure_5a.csv',col.names=T,row.names=T,quote=F)

```

# Perform phylogenetic independence contrasts for this this data

```{r}

# Performing the PIC analysis is surprisingly easy. Data: 

TK_num_vec
Y_freq_vec

# Remove the Hydra species because we cannot seem to find their orthologs in Ensembl Compara (we still do not understand why this is the case).

TK_num_vec <- TK_num_vec[names(TK_num_vec) != 'HVulgaris']
Y_freq_vec <- Y_freq_vec[names(Y_freq_vec) != 'HVulgaris']

# Use the full species names so that we can match the data to the tree

spec_nom <- c('allomyces_macrogynus','coprinopsis_cinerea','saccharomyces_cerevisiae','schizosaccharomyces_pombe','spizellomyces_punctatus',
              'ustilago_maydis','acanthaster_planci','amphimedon_queenslandica','branchiostoma_lanceolatum', 'caenorhabditis_elegans','canis_lupus','capitella_teleta','drosophila_melanogaster','danio_rerio','gallus_gallus','helobdella_robusta',
                        'homo_sapiens','ixodes_scapularis','lottia_gigantea','lingula_anatina','mizuhopecten_yessoensis',
                        'nematostella_vectensis','strongylocentrotus_purpuratus','trichoplax_adhaerens','xenopus_tropicalis')

names(TK_num_vec) <- spec_nom
names(Y_freq_vec) <- spec_nom

# PIC

tre <- ape::read.tree('concat_rooted.treefile')  # Load in the species tree

# Does not work because the tree needs to be rooted for PIC

tk_pic <- ape::pic(TK_num_vec, tre, scaled = TRUE, var.contrasts = FALSE, rescaled.tree = FALSE)
y_pic <- ape::pic(Y_freq_vec, tre, scaled = TRUE, var.contrasts = FALSE, rescaled.tree = FALSE)

# No correlation

pic_df <- data.frame(tk_pic,y_pic)
colnames(pic_df) <- c('TK','Y')

p <- ggplot(pic_df, aes(x=TK, y=Y)) + geom_point(fill='black', stroke=0.001, shape=21, alpha=1, size=4, colour='transparent') 
p <- p+geom_smooth(method=lm, se=FALSE, col='red', lwd=0.6)
p <- p+theme_bw() 
# Ticks
p <- p+theme(axis.ticks.length=unit(-0.10, "cm"), axis.text.x = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")), axis.text.y = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")))
p <- p+theme_bw() + theme(text=element_text(face="plain", size=15), panel.border = element_rect(color="black", size=1.2, linetype="solid"))+theme(legend.position="none")
p <- p+theme(axis.text=element_text(size=9),axis.title.x=element_text(size=11,face="bold"),axis.title.y=element_text(size=11,face="bold"),plot.title=element_text(size=18,face='bold'))
p <- p + xlab('Y kinase contrasts') + ylab('Y content contrasts') + ggtitle('')
p <- p+theme(plot.title = element_text(hjust = 0.5))
p <- p + theme(legend.title=element_text(size=8.5),legend.text = element_text(size=10))
p <- p + geom_richtext(x = 42, y = 0.0023, label = 'r^2 = 0.036', check_overlap = TRUE, size=4.0, label.color = NA)

ggsave(file='Figure_S9d.pdf', plot=p, width=3.0, height=3.0)

```
