---
title: "Figure_S8b"
author: "David Bradley"
date: '2023-08-14'
output: html_document
---

# Read in the files of native pY proteins in yeast:

```{r}

### Native sites

master_all <- read.csv('master_file_unique_new_native.csv')
native_all <- read.table('native_all_Lanz_Leutert_intersect.txt') # Defined as pY found in both Lanz et al and Leutert et al 2022
native_all <- unname(rapply(strsplit(unlist(native_all),split='_'), function(x) x[1]))

####  unique native sites 

accession_vec <- unique(native_all)

```

# Generate a list of spurious pY IDs while filtering out proteins that are only phosphorylated on serine or threonine.

```{r}

pSTY_sub <- read.table('hfsp_unique_substrate_position_IDs.txt') # A list of all unique accessions-positions with the phosphoacceptor position marked.
spurious_accessions <- unique(rapply(strsplit(pSTY_sub[,1],split='_'), function(x) x[1]))

pS_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'S',]
pT_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'T',]
pY_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'Y',]

pS_id <- rapply(strsplit(pS_sub,split='_'), function(x) x[1])
pT_id <- rapply(strsplit(pT_sub,split='_'), function(x) x[1]) 
pY_id <- rapply(strsplit(pY_sub,split='_'), function(x) x[1])

pSpT_acc <- setdiff(c(pS_id,pT_id),pY_id)

# remove such proteins from the accession vector

spurious_accessions <- setdiff(spurious_accessions, pSpT_acc)
accession_vec_spurious <- spurious_accessions

```

# Iterate through all unique proteins that are natively phosphorylated on pY and find the human orthologues

```{r}

vertebrate_ortho <- list.files('') # all files in the directory containing vetrebrate ensembl sequences
vertebrate_ortho <- vertebrate_ortho[file.info(list.files('',full.name=T))$size > 0] # filter for non-empty files

vertebrate_accession <- rapply(strsplit(vertebrate_ortho,split='_'), function(x) x[1])

# Iterate through the native yeast accessions and determine how many have a human orthologue

hs_ortho_vec <- NULL

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  
  if(length(hs_ortho) > 0) {hs_ortho_vec <- c(hs_ortho_vec, yeast_accession)}
    
}

# Repeat but this time for the non-phosphorylated proteins in our dataset (for cases where there is a native pY that we did not detect via MS/MS) 

nonphos_vertebrate_ortho <- list.files('') 
nonphos_vertebrate_ortho <- nonphos_vertebrate_ortho[file.info(list.files('',full.name=T))$size > 0]
nonphos_vertebrate_accession <- rapply(strsplit(nonphos_vertebrate_ortho,split='_'), function(x) x[1])

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(nonphos_vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(nonphos_vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  
  if(length(hs_ortho) > 0) {hs_ortho_vec <- c(hs_ortho_vec, yeast_accession)}
    
}

# ~49.0% of the yeast substrates have at least one orthologue in human

length(unique(hs_ortho_vec))/length(accession_vec)

# Generate simple table of native yeast proteins and their human orthologues. As before, we iterate both through the proteins that were pY-phosphorylated in this study
# and those that were not pY-phosphorylated in this study.

sc_hs_df <- NULL

setwd("")

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  hs_seq <- vertebrate_seq[hs_ortho]
  
  hs_name <- NULL
  sc_name <- NULL
  ortho_class <- NULL
  
  if(length(hs_ortho) == 0) {next}
  
  hs_name <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[1])
  sc_name <- rep(yeast_accession,length(hs_name))
  ortho_class <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[4])
  
  sc_hs <- data.frame(sc_name,hs_name,ortho_class)
  
  sc_hs_df <- rbind(sc_hs, sc_hs_df)
  
}

sc_hs_df_phospho <- data.frame(sc_hs_df,rep('phosphorylated',nrow(sc_hs_df)))
colnames(sc_hs_df_phospho) <- c('sc_name','hs_name','ortho_class','phospho_class')

###

setwd("")

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(nonphos_vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(nonphos_vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  hs_seq <- vertebrate_seq[hs_ortho]
  
  hs_name <- NULL
  sc_name <- NULL
  ortho_class <- NULL
  
  if(length(hs_ortho) == 0) {next}
  
  hs_name <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[1])
  sc_name <- rep(yeast_accession,length(hs_name))
  ortho_class <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[4])
  
  sc_hs <- data.frame(sc_name,hs_name,ortho_class)
  
  sc_hs_df <- rbind(sc_hs, sc_hs_df)
  
}

sc_hs_df_phospho <- data.frame(sc_hs_df,rep('phosphorylated',nrow(sc_hs_df)))
colnames(sc_hs_df_phospho) <- c('sc_name','hs_name','ortho_class','phospho_class')

# Write out the data frame to file:

sc_hs_df_phospho <- data.frame(sc_hs_df,rep('phosphorylated',nrow(sc_hs_df)))
colnames(sc_hs_df_phospho) <- c('sc_name','hs_name','ortho_class','phospho_class')

write.table(sc_hs_df_phospho,'S8a_sc_hs_df_phospho.txt',row.names=F,quote=F)

```

# Now, iterate through the yeast proteins that are NOT phosphorylated and find the vertebrate orthologues (this gives is a baseline expectation for the number of human orthologues that
# we should expect for a random set of proteins of this size)

```{r}

# For the nonphos accessions, the fairest way to do this is to choose only those proteins that we detected in the MS/MS (i.e. from proteome or phosphoproteome)
# Using the whole proteome would introduce several different biases

alex_h_proteome <- read.csv('')
alex_h_unique <- unique(alex_h_proteome[,5])

jan_24_data <- readr::read_csv('')
jan_24_unique <- unique(as.data.frame(jan_24_data)[,4])
nonphos_uniprot <- unique(c(alex_h_unique,jan_24_unique))

# Convert from UniProt to Ensembl

library(clusterProfiler)
library(org.Sc.sgd.db)
uniprot_ensembl <- bitr(nonphos_uniprot, fromType="UNIPROT", toType=c("ENSEMBL"), OrgDb="org.Sc.sgd.db")
nonphos_accessions <- setdiff(uniprot_ensembl[,2],accession_vec_spurious)

# remove native IDs (we conservatively exlcude any protein that is either natively phosphorylated (pY) in the Lanz or Leutert datasets)

native_union <- read.table('native_all_Lanz_Leutert_union.txt')[,1]
native_union <- rapply(strsplit(native_union,split='_'),function(x) x[1])
nonphos_accessions <- setdiff(nonphos_accessions, native_union)

```

# Construct an orthology dataframe for the yeast proteins that are not phosphorylated (this gives is a baseline expectation for the number of human orthologues that
# we should expect for a random set of proteins of this size)

```{r}

setwd("")

nonphos_vertebrate_ortho <- list.files('')  # all files in the directory containing vetrebrate ensembl sequences for the non-phosphorylated proteins 
nonphos_vertebrate_ortho <- nonphos_vertebrate_ortho[file.info(list.files('',full.name=T))$size > 0] # all such files that are not empty
nonphos_vertebrate_accession <- rapply(strsplit(nonphos_vertebrate_ortho,split='_'), function(x) x[1])

# Construct an analogous table for the non-phosphorylated proteins

sc_hs_nophos_df <- NULL

for (i in 1:length(nonphos_accessions)) {
  
  yeast_accession <- nonphos_accessions[i]
  vert_dex <- which(nonphos_vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(nonphos_vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  hs_seq <- vertebrate_seq[hs_ortho]
  
  hs_name <- NULL
  sc_name <- NULL
  ortho_class <- NULL
  
  if(length(hs_ortho) == 0) {next}
  
  hs_name <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[1])
  sc_name <- rep(yeast_accession,length(hs_name))
  ortho_class <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[4])
  
  sc_hs <- data.frame(sc_name,hs_name,ortho_class)
  
  sc_hs_nophos_df <- rbind(sc_hs, sc_hs_nophos_df)
  
}

sc_hs_nophos_df <- data.frame(sc_hs_nophos_df,rep('non-phosphorylated',nrow(sc_hs_nophos_df)))
colnames(sc_hs_nophos_df) <- c('sc_name','hs_name','ortho_class','phospho_class')

# There is no need to write out this table as it is identical to the one generated when performing the Figure 4a analysis.

length(unique(sc_hs_nophos_df[,1]))/length(nonphos_accessions) #47.9%

```

## Process the human phosphorylation data (PSP >= 5)

```{r}

# As a reminder, for the 'high confidence' pY we use all pY that are supported by at least 5 sources in PSP

hs_pY_high_conf <- read.table('human_psp_5.txt',head=T)
hs_pY_high_conf <- hs_pY_high_conf[,c(3,5,10)]
hs_pY_high_conf[,2] <- readr::parse_number(hs_pY_high_conf[,2])
hs_pY_high_conf[,3] <- toupper(hs_pY_high_conf[,3])
hs_pY_high_conf[,1] <- rapply(strsplit(hs_pY_high_conf[,1],split='-'), function(x) x[1]) # Remove the uniprot isoform info as this cannot be mapped to Ensembl IDs

colnames(hs_pY_high_conf) <- c('accession','position','window')

hs_pY_high_conf_id <- unique(hs_pY_high_conf[,1])

# We need to convert these uniprot IDs to Ensembl

library(clusterProfiler)
library(org.Hs.eg.db)
hs_pY_uniprot_ensembl <- bitr(hs_pY_high_conf_id, fromType="UNIPROT", toType=c("ENSEMBL"), OrgDb="org.Hs.eg.db")

# Map back to the original table

hs_pY_high_conf_ensembl <- data.frame(hs_pY_uniprot_ensembl[match(hs_pY_high_conf[,1],hs_pY_uniprot_ensembl[,1]),2],hs_pY_high_conf)
colnames(hs_pY_high_conf_ensembl) <- c('ensembl','accession','position','window')

# Extract the unique IDs
hs_pY_high_conf_ensembl_id <- unique(hs_pY_uniprot_ensembl[,2])

```

# Do the orthology mapping with respect to phosphorylation in human

```{r}

# For proteins that were found to be phosphorylated in our dataset

setwd("")

hs_pY_ortho_vec <- NULL

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_id <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[1])
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho_id <- vertebrate_id[grep('homo_sapiens',vertebrate_species)]
  
  # Check to see if at least one of the orthologues is tyrosine-phosphorylated
  
  hs_pY_dex <- which(hs_ortho_id %in% hs_pY_high_conf_ensembl_id)

  if(length(hs_pY_dex) > 0) {hs_pY_ortho_vec <- c(hs_pY_ortho_vec, yeast_accession)}
    
}

# For proteins that were not found to be phosphorylated in our dataset

setwd("")

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(nonphos_vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(nonphos_vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_id <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[1])
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho_id <- vertebrate_id[grep('homo_sapiens',vertebrate_species)]

  # Check to see if at least one of the orthologues is tyrosine-phosphorylated
  
  hs_pY_dex <- which(hs_ortho_id %in% hs_pY_high_conf_ensembl_id)

  if(length(hs_pY_dex) > 0) {hs_pY_ortho_vec <- c(hs_pY_ortho_vec, yeast_accession)}
    
}

# around ~37.4% of native yeast substrates has at least one orthologue in human that is also phosphorylated on tyrosine

length(unique(hs_pY_ortho_vec))/length(accession_vec)

```

# For yeast proteins with human orthologues, check for enrichment of (natively) phosphorylated yeast proteins against phosphorylated human proteins 

```{r}

sc_hs_df_phospho
sc_hs_nophos_df
  
# sc pY and has an orthologue that is phosphorylated in human 

a <- unique(sc_hs_df_phospho[(sc_hs_df_phospho[,2] %in% hs_pY_high_conf_ensembl_id),1])

# sc pY and has an orthologue that is not phosphorylated in human - must remove proteins that have a different orthologue that is phosphorylated
b <- unique(sc_hs_df_phospho[!(sc_hs_df_phospho[,2] %in% hs_pY_high_conf_ensembl_id),1])
b <- setdiff(b,a)

# sc non-pY and has an orthologue that is phosphorylated in human 

c <- unique(sc_hs_nophos_df[(sc_hs_nophos_df[,2] %in% hs_pY_high_conf_ensembl_id),1])

# sc non-pY and has an orthologue that is not phosphorylated in human 
d <- unique(sc_hs_nophos_df[!(sc_hs_nophos_df[,2] %in% hs_pY_high_conf_ensembl_id),1])
d <- setdiff(d,c)

sc_phos_ortho_phos <- length(a)
sc_phos_ortho_non_phos <- length(b)
sc_non_phos_ortho_phos <- length(c)
sc_non_phos_ortho_non_phos <- length(d)

hs_pY_hc_vec <- sc_phos_ortho_phos

# construct the contingency table

sc_phos_ortho_phos_mat <- matrix(c(sc_phos_ortho_phos,sc_phos_ortho_non_phos,sc_non_phos_ortho_phos,sc_non_phos_ortho_non_phos),ncol=2,byrow=T)

rownames(sc_phos_ortho_phos_mat) <- c('yeast pY (ortho)','yeast non-pY (ortho)')
colnames(sc_phos_ortho_phos_mat) <- c('human pY','human non-pY')

fisher.test(sc_phos_ortho_phos_mat)
fisher.test(t(sc_phos_ortho_phos_mat))

# There is clear enrichment of phosphorylated pY for orthologues with pY (1.3x10-11)

hs_phospho_ortho_vec_random <- (sc_phos_ortho_phos+sc_phos_ortho_non_phos)*(sc_non_phos_ortho_phos/sum(sc_non_phos_ortho_phos,sc_non_phos_ortho_non_phos))

# There is clear enrichment of phosphorylated pY for orthologues with pY (1.2x10-11)

```

# Plot out the data

```{r}

library(viridis)
library(ggsignif)

num_vec <- c(sc_phos_ortho_phos+sc_phos_ortho_non_phos,sc_phos_ortho_phos,hs_phospho_ortho_vec_random)
df <- data.frame(class=c("yeast pY: human", "yeast pY: human pY\n (observed)", "yeast pY: human pY\n (expected)"),
                num=num_vec)

df$class <- factor(df$class, levels = c("yeast pY: human", "yeast pY: human pY\n (observed)", "yeast pY: human pY\n (expected)"))

write.table(df,file='S8B_df.txt',row.names=F,quote=F)

p <- ggplot(df, aes(x=class, y=num, fill = class, width=.5)) + geom_bar(color=NA, lwd=0, stat='identity',alpha=0.4) + scale_fill_manual(values = c(colors()[131],colors()[551],colors()[261]))
p <- p+theme_bw() 
p <- p+theme(axis.text.y=element_text(size=7.5),axis.text.x=element_text(size=6.2),axis.title.x=element_text(size=4.5,face="plain"),axis.title.y=element_text(size=12.5,face="plain"),plot.title=element_text(size=18,face='bold'))
p <- p+theme(axis.ticks.length=unit(-0.10, "cm"), axis.text.x = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")), axis.text.y = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")))
p <- p + ylab("frequency") + xlab("") + ggtitle('')
p <- p + theme(legend.position = "none")
p <- p + theme(panel.border= element_blank())
p <- p + geom_signif(comparisons = list(c("yeast pY: human pY\n (observed)", "yeast pY: human pY\n (expected)")), annotations="***",y_position = 57)
p <- p+ coord_cartesian(ylim = c(0, 147))

ggsave(file='S8b.pdf', plot=p, width=3.4, height=3.4)

```
