---
title: "Figure 4a"
author: "David Bradley"
date: '2023-07-17'
output: html_document
---

# First job is to generate a list of uniquely phosphorylated yeast proteins (from tyrosine kinases):

```{r}

master_files <- list.files('') # Directory of master files for the different kinases

# Loop through all of the master files and retrieve the unique accessions

accession_vec <- NULL

for (i in 1:length(master_files)) {
  
  master_file <- read.csv(paste(master_files[i],sep=''))
  accession_vec <- c(accession_vec, master_file[,7])
  
}

accession_vec <- unique(accession_vec)
accession_vec <- accession_vec[!is.na(accession_vec)]

# Write out the table of unique accessions

write.table(accession_vec,file='hfsp_unique_substrate_IDs.txt',col.names=F,quote=F, row.names=F)

### Native sites

master_all <- read.csv('master_file_unique_new_native.csv') # spreadsheet of all of the unique yeast pY sites
native_all <- read.table('native_all_Lanz_Leutert_union.txt') # spreadsheet of all of the unique native pY sites
native_all <- unname(rapply(strsplit(unlist(native_all),split='_'), function(x) x[1]))

#### remove native 

accession_vec <- setdiff(accession_vec, native_all)

```

# A small number of these accessions have pS/pT sites only and so these need to be filtered out

```{r}

pSTY_sub <- read.table('hfsp_unique_substrate_position_IDs.txt') # A list of all unique accessions-positions with the phosphoacceptor position marked.

pS_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'S',]
pT_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'T',]
pY_sub <- pSTY_sub[substr(rapply(strsplit(pSTY_sub[,1],split='_'), function(x)x[2]),1,1) == 'Y',]

pS_id <- rapply(strsplit(pS_sub,split='_'), function(x) x[1])
pT_id <- rapply(strsplit(pT_sub,split='_'), function(x) x[1]) 
pY_id <- rapply(strsplit(pY_sub,split='_'), function(x) x[1])

pSpT_acc <- setdiff(c(pS_id,pT_id),pY_id)

# remove such proteins from the accession vector

accession_vec <- setdiff(accession_vec, pSpT_acc)
accession_vec_spurious <- accession_vec

```

# Iterate through the yeast proteins and find the vertebrate orthologues using the Ensembl REST-API

```{r}

vertebrate_ortho <- list.files('ensembl_vertebrate_orthologs') # Directory of files containing the orthology data
vertebrate_ortho <- vertebrate_ortho[file.info(list.files('ensembl_vertebrate_orthologs',full.name=T))$size > 0] # Remove files where no orthologs were found

vertebrate_accession <- rapply(strsplit(vertebrate_ortho,split='_'), function(x) x[1])

# Iterate through the yeast accessions and determine how many have a human orthologue

hs_ortho_vec <- NULL

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  
  if(length(hs_ortho) > 0) {hs_ortho_vec <- c(hs_ortho_vec, yeast_accession)}
    
}

# around ~56% of the yeast substrates have an orthologue in human

length(hs_ortho_vec)/length(accession_vec)

# Generate simple table of yeast proteins and their human orthologues

sc_hs_df <- NULL

for (i in 1:length(accession_vec)) {
  
  yeast_accession <- accession_vec[i]
  vert_dex <- which(vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  hs_seq <- vertebrate_seq[hs_ortho]
  
  hs_name <- NULL
  sc_name <- NULL
  ortho_class <- NULL
  
  if(length(hs_ortho) == 0) {next}
  
  hs_name <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[1])
  sc_name <- rep(yeast_accession,length(hs_name))
  ortho_class <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[4])
  
  sc_hs <- data.frame(sc_name,hs_name,ortho_class)
  
  sc_hs_df <- rbind(sc_hs, sc_hs_df)
  
}

sc_hs_df_phospho <- data.frame(sc_hs_df,rep('phosphorylated',nrow(sc_hs_df)))
colnames(sc_hs_df_phospho) <- c('sc_name','hs_name','ortho_class','phospho_class')

write.table(sc_hs_df_phospho,'sc_hs_df_phospho.txt',row.names=F,quote=F)

```

# Now, iterate through the yeast proteins that are NOT phosphorylated and find the vertebrate orthologues

```{r}

accession_vec <- accession_vec

# Read in proteome from SGD (we need the SGD proteome because of the systematic locus names that are not present in the UniProt files)

sc_proteome <- seqinr::read.fasta('orf_trans_all_R64-3-1_20210421.fasta')
nonphos_accessions <- setdiff(names(sc_proteome), accession_vec)

# Retrieve vertebrate orthologues and sequences

library(httr)
library(jsonlite)
library(xml2)
library(ggplot2)
library(seqinr)

for (i in 1:length(nonphos_accessions)) {  #

  print(i)
  
  yeast_id <- nonphos_accessions[i]
    
  # Retrieve fungal orthologues for each S. cerevisae SH3
    
  server <- "https://rest.ensembl.org"
  ext <- paste("/homology/id/",yeast_id,"?aligned=0;compara=vertebrates",sep="")
  httr::set_config(httr::config(ssl_cipher_list = 'DEFAULT@SECLEVEL=1'))
  r <- GET(paste(server, ext, sep = ""), content_type("application/json"))
     
  stop_for_status(r)
     
  # Parse the JSON object
    
  orthology_df <- fromJSON(toJSON(content(r)),simplifyDataFrame = TRUE)[[1]]$homologies[[1]]
  
  # Skip if there are no orthologues
  
  if(is.null(orthology_df)) {next}
  if(length(orthology_df) == 0) {next}
  
  taxonomy <- unlist(orthology_df[,colnames(orthology_df) == 'taxonomy_level'])
  homolog_class <- unlist(orthology_df[,which(colnames(orthology_df) == 'type')])
  orthology_all <- orthology_df[which(unlist(orthology_df[,which(colnames(orthology_df) == 'type')]) %in% c('ortholog_one2one','ortholog_one2many','ortholog_many2many')),]
  taxonomy <- taxonomy[which(unlist(orthology_df[,which(colnames(orthology_df) == 'type')]) %in% c('ortholog_one2one','ortholog_one2many','ortholog_many2many'))]
  
  orthology_all_type <- orthology_all[,which(colnames(orthology_all)=='target')]
  
  orthology_all_seq <- orthology_all_type[,which(colnames(orthology_all_type) == 'seq')]
  orthology_all_seq_species <- unlist(orthology_all_type[,which(colnames(orthology_all_type) == 'species')])
  orthology_all_id <- unlist(orthology_all_type[,which(colnames(orthology_all_type) == 'id')])
  orthology_category <- homolog_class[which(homolog_class %in% c('ortholog_one2one','ortholog_one2many','ortholog_many2many'))]
  
  if(length(orthology_category) != length(orthology_all_seq)) {stop()}
  
  # Generate unique strings for duplicate names
  orthology_all_seq_species <- make.names(orthology_all_seq_species,unique=T)
  orthology_all_seq_id_species <- paste(orthology_all_id,':',orthology_all_seq_species,sep='')
  orthology_all_seq_id_species_taxonomy <- paste(orthology_all_seq_id_species,':',taxonomy,sep='')
  orthology_all_seq_id_species_taxonomy_class <- paste(orthology_all_seq_id_species_taxonomy,':',orthology_category,sep='')
  
  # Write out the orthologue sequences
    
  file_name <- paste(yeast_id,'_nonphos_compara_ortholog_vertebrates.fa',sep='')
  file_name <- paste('/home/david/Documents/Work/HFSP_evolution/ensembl_nonphos_vertebrate_orthologs/',file_name,sep='')
  write.fasta(c(orthology_all_seq),file.out=file_name,names=c(orthology_all_seq_id_species_taxonomy_class))

}  

# For the nonphos accessions it is actually a fairer test to choose only those proteins that we detected in the MS/MS (i.e. from proteome or phosphoproteome)
# Using the whole proteome would introduce several different biases

alex_h_proteome <- read.csv('data_export_protein.csv') #proteome data
alex_h_unique <- unique(alex_h_proteome[,5])

jan_24_data <- readr::read_csv('statistical_test_export.csv') #phosphorylation data
jan_24_unique <- unique(as.data.frame(jan_24_data)[,4])
nonphos_uniprot <- unique(c(alex_h_unique,jan_24_unique))

# Convert from UniProt to Ensembl

library(clusterProfiler)
library(org.Sc.sgd.db)
uniprot_ensembl <- bitr(nonphos_uniprot, fromType="UNIPROT", toType=c("ENSEMBL"), OrgDb="org.Sc.sgd.db")
nonphos_accessions <- setdiff(uniprot_ensembl[,2],accession_vec) # Remove phosphorylated proteins from the set of all yeast proteins that were detected during mass spectrometry (phosphorylated or non-phosphorylated)

# remove native

nonphos_accessions <- setdiff(nonphos_accessions, native_all)

```

# Construct an orthology dataframe for the yeast proteins that are NOT phosphorylated

```{r}

nonphos_vertebrate_ortho <- list.files('ensembl_nonphos_vertebrate_orthologs') # Directory of files containing the orthology data for non-phosphorylated proteins 
nonphos_vertebrate_ortho <- nonphos_vertebrate_ortho[file.info(list.files('ensembl_nonphos_vertebrate_orthologs',full.name=T))$size > 0] # Remove files where no orthologs were found
nonphos_vertebrate_accession <- rapply(strsplit(nonphos_vertebrate_ortho,split='_'), function(x) x[1])

# Construct an analogous table for the non-phosphorylated proteins

sc_hs_nophos_df <- NULL

for (i in 1:length(nonphos_accessions)) {
  
  yeast_accession <- nonphos_accessions[i]
  vert_dex <- which(nonphos_vertebrate_accession == yeast_accession)
  
  #Skip if there are no vertebrate orthologues
  if(length(vert_dex) == 0) {next}
  
  vertebrate_seq <- seqinr::read.fasta(nonphos_vertebrate_ortho[vert_dex],seqtype = 'AA')
  vertebrate_species <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[2])
  vertebrate_class <- rapply(strsplit(names(vertebrate_seq),split='\\:'),function(x) x[4])
  
  # Find if there is a human orthologue present
  
  hs_ortho <- grep('homo_sapiens',vertebrate_species)
  hs_seq <- vertebrate_seq[hs_ortho]
  
  hs_name <- NULL
  sc_name <- NULL
  ortho_class <- NULL
  
  if(length(hs_ortho) == 0) {next}
  
  hs_name <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[1])
  sc_name <- rep(yeast_accession,length(hs_name))
  ortho_class <- rapply(strsplit(names(hs_seq),split=':'), function(x) x[4])
  
  sc_hs <- data.frame(sc_name,hs_name,ortho_class)
  
  sc_hs_nophos_df <- rbind(sc_hs, sc_hs_nophos_df)
  
}

sc_hs_nophos_df <- data.frame(sc_hs_nophos_df,rep('non-phosphorylated',nrow(sc_hs_nophos_df)))
colnames(sc_hs_nophos_df) <- c('sc_name','hs_name','ortho_class','phospho_class')

```

# Fisher test of orthology enrichment

```{r}

sc_hs_ortho_all <- rbind(sc_hs_df_phospho,sc_hs_nophos_df)

# sc pY and has an orthologye

sc_phos_ortho <- length(which(accession_vec %in% sc_hs_ortho_all[,1]))
sc_phos_non_ortho <- length(which(!accession_vec %in% sc_hs_ortho_all[,1]))
sc_non_phos_ortho <- length(which(nonphos_accessions %in% sc_hs_ortho_all[,1]))
sc_non_phos_non_ortho <- length(which(!nonphos_accessions %in% sc_hs_ortho_all[,1]))

# construct the contingency table

sc_phos_ortho_mat <- matrix(c(sc_phos_ortho,sc_phos_non_ortho,sc_non_phos_ortho,sc_non_phos_non_ortho),ncol=2)

rownames(sc_phos_ortho_mat) <- c('ortho','non-ortho')
colnames(sc_phos_ortho_mat) <- c('pY','not pY')

fisher.test(sc_phos_ortho_mat)
fisher.test(t(sc_phos_ortho_mat))

# Very highly significant

# Conclusion: proteins that are spuriously phosphorylated are more likely to have a human orthologue than non-phosphorylated proteins (i.e.
# spurious interactions are biased towards interactions with homologues of native interaction partners)

# The random expectation therefore for the number of pY with orthologues is simply the number of pY protein *(non-pY / non-pY with orthologues)

hs_ortho_vec_random <- length(accession_vec)*(sc_non_phos_ortho/sum(sc_non_phos_ortho,sc_non_phos_non_ortho))

```

# Figure 4A GGplot

```{r}

library(viridis)
library(ggsignif)
library(ggplot2)

num_vec <- c(length(accession_vec),length(hs_ortho_vec),hs_ortho_vec_random)
df <- data.frame(class=c("yeast pY", "yeast pY: human \n (observed)", "yeast pY: human \n (expected)"),
                num=num_vec)

df$class <- factor(df$class, levels = c("yeast pY", "yeast pY: human \n (observed)", "yeast pY: human \n (expected)"))

p <- ggplot(df, aes(x=class, y=num, fill = class, width=.5)) + geom_bar(color=NA, lwd=0, stat='identity',alpha=0.4) + scale_fill_manual(values = c(rev(cividis(25))[1],colors()[131],colors()[261]))
p <- p+theme_bw() 
p <- p+theme(axis.text.y=element_text(size=7),axis.text.x=element_text(size=5.75),axis.title.x=element_text(size=4.5,face="plain"),axis.title.y=element_text(size=11.5,face="plain"),plot.title=element_text(size=10,face='plain'))
p <- p+theme(axis.ticks.length=unit(-0.10, "cm"), axis.text.x = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")), axis.text.y = element_text(margin=unit(c(0.2,0.2,0.2,0.2), "cm")))
p <- p + ylab("frequency") + xlab("") + ggtitle('conservation (protein level)')
p <- p+theme(plot.title = element_text(hjust = 0.5))
p <- p + theme(legend.position = "none")
p <- p + theme(panel.border= element_blank())
p <- p + geom_signif(comparisons = list(c("yeast pY: human \n (observed)", "yeast pY: human \n (expected)")), annotations="***",y_position=790)

ggsave(file='Figure_4A.pdf', plot=p, width=3.0, height=3.0)

```


